common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapConfigStoreTaskProvider.java

```diff
@@ -164,9 +166,20 @@ protected void findSecurityInstanceId(Link[] objectLinks) {
 
     protected void findServerInstanceId(BootstrapReadResponse readResponse) {
         this.serverInstances = new HashMap<>();
-        ((LwM2mObject) readResponse.getContent()).getInstances().values().forEach(instance -> {
-            serverInstances.put(((Long) instance.getResource(0).getValue()).intValue(), instance.getId());
-        });
+        try {
```

Comment:
What is the scope of these changes and why it is not listed in the issue description? I really don't understand what is this about.

Reply:
f there is a model (xml format) an integer number will come
If there is no model, bytes will come (bytes - the size of the unknown array, depends on the number)
Tested on this data
				<Name>Short Server ID</Name>
        <Operations>R</Operations>
				<MultipleInstances>Single</MultipleInstances>
				<Mandatory>Mandatory</Mandatory>
				<Type>Integer</Type>
				<RangeEnumeration>1..65534</RangeEnumeration>

Reply:
I am merging this, but LwM2MBootstrapConfigStoreTaskProvider is far from perfect. The method "getTasks" is not thread-safe because it stores temporary results of the execution somewhere in the object variables.

Reply:
This issue is not within the scope of this PR.